packages/scenario/test/relevant.test.ts

```diff
@@ -0,0 +1,223 @@
+		 *   implied by this, discussed in greater detail in porting notes on
+		 *   {@link UntypedAnswer}.
+		 *
+		 * Two additional variants of this test are added immediately following this
+		 * one, both briefly exploring some of the contours of the current failure.
+		 */
+		it.fails('is inherited from ancestors', async () => {
+			const scenario = await Scenario.init(
+				'Some form',
+				html(
+					head(
+						title('Some form'),
+						model(
+							mainInstance(
+								t(
+									'data id="some-form"',
+									t('is-group-relevant'),
+									t('is-field-relevant'),
+									t('group', t('field'))
+								)
+							),
+							bind('/data/is-group-relevant').type('boolean'),
+							bind('/data/is-field-relevant').type('boolean'),
+							bind('/data/group').relevant('/data/is-group-relevant'),
+							bind('/data/group/field').type('string').relevant('/data/is-field-relevant')
+						)
+					),
+					body(
+						input('/data/is-group-relevant'),
+						input('/data/is-field-relevant'),
+						group('/data/group', input('/data/group/field'))
+					)
+				)
+			);
+
+			// Form initialization evaluates all triggerables, which makes the group and
+			//field non-relevants because their relevance expressions are not satisfied
+			expect(scenario.getAnswerNode('/data/group')).toBeNonRelevant();
+			expect(scenario.getAnswerNode('/data/group/field')).toBeNonRelevant();
+
+			// Now we make both relevant
+			scenario.answer('/data/is-group-relevant', true);
+			scenario.answer('/data/is-field-relevant', true);
+
+			expect(scenario.getAnswerNode('/data/group')).toBeRelevant();
+			expect(scenario.getAnswerNode('/data/group/field')).toBeRelevant();
+
+			// Now we make the group non-relevant, which makes the field non-relevant
+			// regardless of its local relevance expression, which would be satisfied
+			// in this case
+			scenario.answer('/data/is-group-relevant', false);
+
+			expect(scenario.getAnswerNode('/data/group')).toBeNonRelevant();
+			expect(scenario.getAnswerNode('/data/group/field')).toBeNonRelevant();
+		});
+
+		/**
+		 * **PORTING NOTES** (first variant of ported test above)
+		 *
+		 * This test is identical to the test above, except that both `relevant`
+		 * expressions are wrapped in a `string()` XPath call. The test still fails,
+		 * but notably the failing assertion comes later:
+		 *
+		 * In the original test, the first assertion fails because a `node-set`
+		 * expression which resolves to any node will always cast to `true`. When
+		 * the value is cast to a string, the node's text value is consulted in
+		 * casting, producing `false` when empty.
+		 *
+		 * Ultimately, the test fails when checking restoration of the `false`
+		 * state. This is because the `false` value is presently being persisted to
+		 * the primary instance as the string `"0"` (which, as I recall, is the
+		 * expected serialization of boolean `false`). Since the `relevant`
```

Comment:
As currently implemented in `@getodk/xpath`, I am 98% certain that we treat `"0"`-as-falsy consistently with the casting rules in XPath 1.0. I suspect there will be some edge case discrepancies on interpretation between JR/WF on this regardless of the other contributing factors (e.g. whether we end up supporting a `boolean` bind type).

Re: `"false", I think we'll want to take an approach similar to the one I took to support the various `date` casting cases. Ultimately that boiled down to:

- overriding the `number` function (additive override, defining the function with the same local name but in the XForms namespace)

- making sure certain casting code paths invoke whichever `number` function is resolved for the active `Evaluator` instance (I don't recall offhand how many places this happens, but if it's more than one it's not many)

This would definitely deviate from XPath 1.0. But it feels right to align with ODK/Collect/JR on this, and it feels right to take a similar approach for similar semantic specialization concerns like this.

Reply:
This has been kicking around in the back of my mind. It's only Enketo that serializes boolean values to "false" and "true." We don't expose the `boolean` type in XLSForm. Maybe Web Forms doesn't need to handle them? I think I need to keep think about this.

Reply:
I think this also falls into the probably-less-effort-to-do-than-doubt category, FWIW. I'd be happy to treat stuff like this as low priority rather than out of scope.